libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -1480,6 +1665,30 @@ class MapboxRouteLineApi(
                 RouteLineExpressionProvider { exp }
             }
 
+        val maskingLayerData = if ((primaryRoute?.directionsRoute?.legs()?.size ?: 0) > 1) {
+            activeLegIndex = if (activeLegIndex == INVALID_ACTIVE_LEG_INDEX) {
+                0
+            } else {
+                activeLegIndex
+            }
```

Comment:
We should never end up in this branch of code. Shouldn't we just do `activeLegIndex = 0`? Is there a use case that needs to be considered?

Reply:
There might have been at one time or perhaps I was being conservative. It doesn't appear to be needed in the latest iteration.

Reply:
There might have been at one time or perhaps I was being conservative. It doesn't appear to be needed in the latest iteration.